Week1/index.js

```diff
@@ -0,0 +1,173 @@
+                        indep_year int,
+                        population int,
+                        life_expectancy float,
+                        gnp float,
+                        gnp_old float,
+                        local_name varchar(254),
+                        government_form varchar(254),
+                        head_of_state varchar(254),
+                        capital int,
+                        primary key(id),
+                        foreign key(capital) references city(id)
+                      )`,
+    message: '"country" table created.',
+  }, // createTableCountry
+];
+
+function showErrorAndExit(errorMessage, connection) {
+  console.error(errorMessage);
+  if (connection) connection.end();
+  process.exit();
+}
+
+// Make the connection to the mysql server
+connection.connect(err => {
+  if (err) {
+    showErrorAndExit(`Connection Error: ${err.message} (${err.code})`);
+  }
+  console.log('Successfully connected to mysql server...');
+});
+
+async function createDBAndTables() {
+  for (const query of queries) {
+    try {
+      await asyncQuery(query.query);
+      console.log(query.message);
+    } catch (err) {
+      if (err) {
+        showErrorAndExit(`Query Error: ${err.message} (${err.code})`, connection);
+      }
+    }
+  }
+}
+
+async function insertRecord(records, table) {
+  try {
+    for (const record of records) {
+      await asyncQuery('insert into ' + table + ' SET ?', record);
+      /**
+       * To increase user experience;
+       * Just logs the same line :: \033[0G => \r
+       * console.log creates new line
+       * so process.stdout.write() with '\033[0G' code at the end makes sense for this purpose.
+       */
+      process.stdout.write('Adding into ' + table + ' table. \033[0G');
+    }
+    console.log(`Records added successfully into ${table} table.`);
+  } catch (err) {
+    if (err) {
+      showErrorAndExit(`Query Error: ${err.message} (${err.code})`, connection);
+    }
+  }
+}
+
+// Populate tables with related data
+async function readAndInsertDataIntoTables() {
+  try {
+    // Read & Parse Data
+    const { cities, countries } = JSON.parse(await readFile('./data.json', 'utf-8'));
+
+    // Insert each city record to the city table
+    await insertRecord(cities, 'city');
+
+    // Insert each country record to the country table
+    await insertRecord(countries, 'country');
+
+    // Before exit, close the connection
+    connection.end();
+  } catch (error) {
+    showErrorAndExit(`File Read Error: ${error.message}`, connection);
+  }
+}
+
+async function queryAndShowResultOf(filter, table, fields) {
+  const query = `select ${
+    fields.length > 1 ? fields.join(',') : fields[0]
+  } from ${table} ${filter}`;
+  try {
+    const result = await asyncQuery(query);
+    console.log('\n\nResult of ', filter, '\n');
+    console.table(result);
+  } catch (err) {
+    showErrorAndExit(`Filtering Error on (${filter}): ${err.message}`, connection);
+  }
+}
```

Comment:
I love you tried to avoid repetition in your code, but what do you think is the best: 
 - People can read the code easily and we have some repetition
 - Or you don't have any repetition but to understand the query you have to log it.

Personally I'm the first kind of developer. I always try to avoid repetition but sometimes it's better to keep them for code readability.